### ISSUE TITLE
My site with jspanel and accessibility

### ISSUE THREAD

**Issue by AviHafner** [Flyer53/jsPanel4#45]
Hi Stefan
When I am using the jsPanel.modal
The panel disables the main window from mouse clicking,
 but not from taping on the keyboard (tab).

I wonder if it possible to protect the b main window from taping on the keyboard as well.

You can check the example in 
http://www.admin.technion.ac.il/avi/

Yours
Avi Hafner

**Comment by Flyer53** [comment]
Hi Avi,
to be frank ... I really didn't think about this before.

Try the following approach which could work for you:
```javascript
function blockKeyboard(event) {
  event.preventDefault();
}

jsPanel.modal.create({
  callback: function () {
    document.addEventListener('keydown', blockKeyboard);
  },
  onclosed: function() {
    document.removeEventListener('keydown', blockKeyboard);
  }
});
```
Best Regards,
Stefan 😄

**Comment by Flyer53** [comment]
@AviHafner 
Or if you want to block the Tab key only:
```javascript
function blockTabkey(event) {
  if (event.key === 'Tab') {
    event.preventDefault();
  }  
}

jsPanel.modal.create({
  callback: function () {
    document.addEventListener('keydown', blockTabkey);
  },
  onclosed: function() {
    document.removeEventListener('keydown', blockTabkey);
  }
});
```

**Comment by AviHafner** [comment]
Sorry Stefan
I need to protect the keyword event only from the main window,
But for accessibility :  All the remainder like the panel need to be accessible for the mouse and the keyboard.

yours 
Avi

**Comment by Flyer53** [comment]
Ok Avi ... but what do you mean with **keyword** event?

**Comment by AviHafner** [comment]
Yes you are right, 
We are trying to prepare our sites according to accessibility standards
And people with visual impairments are supposed to use the keyboard, not just the mouse

**Comment by AviHafner** [comment]
-**the keyword event**_? 
all the keyword event that working in the main window,
In the meantime I found only the tab , and the backtab(shift+tab).

**Comment by AviHafner** [comment]
Wow Stephan
I'm getting lost,
it's Not a good news for me

**Comment by Flyer53** [comment]
lol ... you know I‘m always willing to help Avi. l just do have a problem sometimes to understand what your actual problem is. So if you can explain it in a way I understand I might be able to do something for you.

**Comment by AviHafner** [comment]
Thanks Stefan

Is there a possibility when the jsPanel opens
The elements in the main window will not be accessible.

**Comment by Flyer53** [comment]
Good morning Avi,
If you want to block all interaction with the main page until the modal is closed the first of the 2 code samples above should do it.

Btw: [Here](https://www.nomensa.com/blog/2014/how-improve-accessibility-overlay-windows-part-1) is a intersting article dealing with the accessibility topic.

**Comment by Flyer53** [comment]
@AviHafner 
Thinking about this a bit I came to the conclusion that the above approach is probably not the best. Especially if your modal contains a form.

I created a additional example in the docs pages with a different approach. It simply locks the keyboard navigation via the "Tab" key to the modal panel until the modal panel is closed.
https://jspanel.de/api.html#extensions/modal/example-3

I think this might be a good solution for you ...

Stefan :)
